cinnabar/cmd/fetch.py

```diff
@@ -43,7 +43,9 @@ def fetch(args):
                       'revision.',
                       file=sys.stderr)
                 return 1
-        full_revs.append(rev.decode('ascii'))
+        if sys.version_info[0] == 2:
+            rev = rev.decode('ascii')
```

Comment:
This is a no-op on python2. The issue is probably not, in fact, a python2 vs. python3 problem, but rather related to whether you went through the branch above or not. I suspect if you pass an abbreviated sha1 (remove a few characters from it), git cinnabar fetch works without the patch, and breaks with it.

Reply:
yes, indeed.  abbrev sha works.
will fix it to call decode only for abbrev case